### DISCUSSION TITLE
Title Bar Color Doesn't Match Background Color When Transparency Is Applied

### DISCUSSION THREAD

**Discussion by yuki6ra** [ghostty-org/ghostty#3655] [Issue Triage] [answered]
Hi,
First, I’d like to apologize for any mistakes in my English, as I’m using a translation tool.
After asking about this issue on Discord, I was encouraged to bring it here for further discussion.

---

When the `macos-titlebar-style` is set to tabs and the background transparency (`background-opacity`) is less than 1, the title bar color does not match the background color.

Here are my current settings:
```
background-opacity = 0.7  
background-blur-radius = 6  
macos-titlebar-style = tabs  
window-padding-x = 20  
window-padding-y = 0,20  
window-padding-balance = true  
window-theme = auto  
window-height = 50  
window-width = 150  

theme = rose-pine-moon  

keybind = cmd+x=close_surface  
```

image: `background-opacity = 1`
![image](https://github.com/user-attachments/assets/ad29f9f8-5003-40b8-a482-b41a5cb24754)

image: `background-opacity = 0.7`
(Notice that the tab color is darker than the background color!)
![image](https://github.com/user-attachments/assets/aa448a44-6e48-4bc1-a5f6-464c5c48dcd2)

I also tried the following configurations:
- `window-theme = ghostty`: Only the title bar text color changed to black.
- `macos-titlebar-style = native`: The colors still didn’t match.

So For now, I’ve decided to keep the transparency at 100% when using the tabs style.

---

It would be great if the tabs style could match the background color even when transparency is applied. This would improve the visual consistency and make the feature more versatile.

Thank you for considering this!

**Comment by mitchellh** [comment, marked as answer]
At some point this was resolved, try latest tip. Your reproduction looks good on my system! Thank you
